### PR TITLE
Added PromoBanner analytics, removed old

### PR DESCRIPTION
**Pull request by bsmartin-ep**
## Description

This PR will:

1. Update component-library package version
2. Add PromoBanner link click (aka close) to taxonomy to lookup object
3. Remove old event taxonomy from _/src/platform/site-wide/announcements/components/PromoBanner.jsx_. I do not see this event firing in GA since March 2020, so we should be OK to remove.

https://github.com/department-of-veterans-affairs/va.gov-team/issues/14497
https://github.com/department-of-veterans-affairs/component-library/pull/60

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/site-wide/announcements/components/PromoBanner.jsx [hunk 1]
```diff
@@ -1,22 +1,7 @@
...
-      onClose={() => {
-        recordEvent({
-          event: 'nav-promo-banner-link',
-        });
-        onClose();
-      }}
-    />
-  );
+export default function PromoBannerWithAnalytics({ ...props }) {
+  return <PromoBanner {...props} />;
```
Comment (ncksllvn) [inline src/platform/site-wide/announcements/components/PromoBanner.jsx:6]: Can we just delete this whole component (now that it is abstracting nothing), and replaces instances of it with the `component-library/PromoBanner`?

FSYA that this probably can be removed too, `src/platform/site-wide/announcements/tests/components/PromoBanner.unit.spec.jsx`

Comment (bsmartin-ep) [reply src/platform/site-wide/announcements/components/PromoBanner.jsx]: Yes indeed @ncksllvn.  Component and test have been removed.

Any suggestions on how to test this adequately when we have no current PromoBanner components in use?  I added one to my local Facility Locator and it seems to be working...

image

File: yarn.lock [hunk 2]
```diff
@@ -19382,6 +19445,14 @@ use-sidecar@^1.0.1:
...
     tslib "^1.9.3"
 
+use-sidecar@^1.0.5:
+  version "1.0.5"
+  resolved "https://registry.yarnpkg.com/use-sidecar/-/use-sidecar-1.0.5.tgz#ffff2a17c1df42e348624b699ba6e5c220527f2b"
+  integrity sha512-k9jnrjYNwN6xYLj1iaGhonDghfvmeTmYjAiGvOr7clwKfPjMXJf4/HOr7oT5tJwYafgp2tG2l3eZEOfoELiMcA==
+  dependencies:
+    detect-node-es "^1.1.0"
+    tslib "^1.9.3"
+
```
Comment (ncksllvn) [inline yarn.lock:19455]: Wow all of these Yarn updates are from the component-library?

File: src/platform/site-wide/announcements/components/PromoBanner.jsx (hunk 1 above)
Comment (ncksllvn) [reply src/platform/site-wide/announcements/components/PromoBanner.jsx]: Sorry I missed this - but exactly how you did it is how I would've tested it too

**Review by ncksllvn — APPROVED** [review]

**Review by jonwehausen — APPROVED** [review]
LGTM

**Comment by bsmartin-ep** [comment]
Hello @department-of-veterans-affairs/frontend-review-group.   Can I get a review of this PR at your leisure?

Thanks!

**Review by U-DON — APPROVED** [review]

**Pull request merged**